Задача 4. Есть ли статистически значимые различия в росте
дочерей?
Рост матерей 172, 177, 158, 170, 178, 175, 164, 160, 169, 165
Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160, 163

In [5]:
import numpy as np
from scipy import stats

In [6]:
sample_0 = np.array([172, 177, 158, 170, 178, 175, 164, 160, 169, 165]) # рост матерей
sample_1 = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160, 163]) # рост дочерей

Нулевая гипотеза усверждает, что статистически значимых различий в росте дочерей и матерей нет.
Альтернативная гипотеза усверждает, что статистически значимые различия в росте дочерей и матерей есть.
Принимаем α=0,05.

Так как дисперсия неизвестна , используем t – критерий.
Так как у нас две выборки, будем использовать двухвыборочный тест для зависимых выборок.

In [7]:
stats.ttest_rel(sample_0, sample_1)

TtestResult(statistic=0.6648478531431979, pvalue=0.5228168632983574, df=9)

Т.к. pvalue=0.5228168632983574 > α=0,05, принимаем нулевую гипотезу (статистически значимых различий в росте дочерей и матерей нет)